Added Firefox install and Firefox test run with the current tests (#311)
iiab/calibre-web: README.md

### Prerequisites

1. **Install Chrome and/or Firefox to enable Selenium testing**:

>    Follow the instructions on this page to [install Chrome for Ubuntu](https://linuxcapable.com/install-google-chrome-on-ubuntu-linux/)

Reviewer: Suggest changing to: 
    - [Install Chrome for Ubuntu](https://linuxcapable.com/install-google-chrome-on-ubuntu-linux/) and/or

   For Firefox, follow these [instructions](https://support.mozilla.org/en-US/kb/install-firefox-linux#w_install-firefox-deb-package-for-debian-based-distributions-recommended)

>    **Note: If you are on Ubuntu Deskop you must remove the Firefox Snap with `sudo snap remove firefox`, or else the tests will fail**

Reviewer: - [Install Firefox](https://support.mozilla.org/en-US/kb/install-firefox-linux#w_install-firefox-deb-package-for-debian-based-distributions-recommended) [NB: If you are on Ubuntu Deskop you must remove the Firefox Snap with `sudo snap remove firefox`, or else the tests will fail]

> **Note: Please run the remaining commands as a non-root user**
> 2. **Install dependencies**:

Reviewer 2: ```suggestion
**Note: Please run the remaining commands as a non-root user**

2. **Install dependencies**:
```

And another very minor suggestion:

Renumber 4., 5., 6. and 7. below?  (e.g. to 3., 4., 5. and 6.)

### Prerequisites

1. **Install Chrome and/or Firefox to enable Selenium testing**:

    - [Install Chrome for Ubuntu](https://linuxcapable.com/install-google-chrome-on-ubuntu-linux/) and/or

    - [Install Firefox](https://support.mozilla.org/en-US/kb/install-firefox-linux#w_install-firefox-deb-package-for-debian-based-distributions-recommended) [NB: If you are on Ubuntu Deskop you must remove the Firefox Snap with `sudo snap remove firefox`, or else the tests will fail]

   **Note: If you are on Ubuntu Deskop you must remove the Firefox Snap with `sudo snap remove firefox`, or else the tests will fail**

**Note: Please run the remaining commands as a non-root user**\n

2. **Install dependencies**:

   Set up the python virtual environment using the following commands:

   ```bash

>    cd /opt/iiab/calibre-web-py3    # cd into the directory where you have Calibre-Web cloned

Reviewer 2: - I'm hoping we can emphasize `/usr/local/calibre-web-py3` not `/opt/iiab/calibre-web-py3` ?

- Somehow at the very top of the instructions — let's spell it out — so that instructions are crystal clear with zero ambiguity on how to proceed (1) with [IIAB Calibre-Web installed](https://github.com/iiab/calibre-web/wiki#wrench-installation) &/or (2) without IIAB Calibre-Web installed.

Author: We don't have a way to know where someone cloned the repo. Can you give an example of what you are exactly looking for?

Author: I added the correct directory and put a small line about how /usr/local/calibre-web-py3 is where IIAB clones the repo. Let me know if you think that is sufficient.

   python3 -m venv calibre-web-env
   source calibre-web-env/bin/activate
   pip install -r requirements.txt
   ```

3. Add the dummy database from IIAB project:

   ```bash

>    wget -O app.db https://github.com/iiab/iiab/raw/refs/heads/master/roles/calibre-web/files/app.db

Reviewer 3: Just curious, is there a reason to avoid the double quotes?

I always use, just in case a space or special character is between the string. 

The reason to remove, is because a harcoded string?

Author: I don't think the quotes were removed on purpose(Adam, Avni and I have all made multiple changes to the README so I am not sure who removed them). I will add them back in.

Reviewer 2: Yes exactly; that's all.

(Quotes are helpful if a softcoded URL might include strange characters.)

   ```

4. Execute Calibre-Web in background:

>    **Note: If you have installed Callibre-Web through IIAB, then this command is not required.**

Reviewer 2: ```suggestion
   **Note: If you have installed Calibre-Web through IIAB, then this command is not required.**
```

   ```bash
   nohup python3 cps.py &
   ```

> 5. Install tests requirements in the same virtual environment as step 3:

Reviewer 2: There's just 1 venv being discussed in this recipe, isn't that right?

```suggestion
5. Install tests requirements in the same virtual environment:
```

Author: You're right, I was just trying to make it as clear as possible for someone who is not familiar with Python environment stuff.

### Prerequisites

1. **Install Chrome and/or Firefox to enable Selenium testing**:

>    Follow the instructions on this page to [install Chrome for Ubuntu](https://linuxcapable.com/install-google-chrome-on-ubuntu-linux/)

Reviewer: ```suggestion
    - [Install Chrome for Ubuntu](https://linuxcapable.com/install-google-chrome-on-ubuntu-linux/) and/or
```

>    For Firefox, follow these [instructions](https://support.mozilla.org/en-US/kb/install-firefox-linux#w_install-firefox-deb-package-for-debian-based-distributions-recommended)

Reviewer: ```suggestion
    - [Install Firefox](https://support.mozilla.org/en-US/kb/install-firefox-linux#w_install-firefox-deb-package-for-debian-based-distributions-recommended) [NB: If you are on Ubuntu Deskop you must remove the Firefox Snap with `sudo snap remove firefox`, or else the tests will fail]
```

   **Note: If you are on Ubuntu Deskop you must remove the Firefox Snap with `sudo snap remove firefox`, or else the tests will fail**

**Note: Please run the remaining commands as a non-root user**\n

2. **Install dependencies**:

>    Ensure you have all required dependencies installed.

Reviewer: ```suggestion
```

>    Then, set up the python virtual environment using the following commands:

Reviewer: ```suggestion
   Set up the python virtual environment using the following commands:
```
